Origin: http://bugs.exim.org/attachment.cgi?id=586
Bug: http://bugs.exim.org/show_bug.cgi?id=1287
Bug-Debian: http://bugs.debian.org/686495
Description: Fix for upstream bug 1287 - wrong value in re_nsub 
 This is due to a wrong cast. Bug fix from upstream bugzilla.
Index: pcre3-8.31/pcreposix.c
===================================================================
--- pcre3-8.31.orig/pcreposix.c	2012-06-20 16:08:49.000000000 +0100
+++ pcre3-8.31/pcreposix.c	2012-09-13 19:53:34.000000000 +0100
@@ -259,6 +259,7 @@
 int erroffset;
 int errorcode;
 int options = 0;
+int re_nsub = 0;
 
 if ((cflags & REG_ICASE) != 0)    options |= PCRE_CASELESS;
 if ((cflags & REG_NEWLINE) != 0)  options |= PCRE_MULTILINE;
@@ -282,7 +283,8 @@
   }
 
 (void)pcre_fullinfo((const pcre *)preg->re_pcre, NULL, PCRE_INFO_CAPTURECOUNT,
-  &(preg->re_nsub));
+  &re_nsub);
+preg->re_nsub = (size_t)re_nsub;
 return 0;
 }
 
